packages/admin/admin-generator/src/commands/generate/generate-command.ts

```diff
@@ -115,11 +108,23 @@ export type FormFieldConfig<T> = (
                     gqlName?: string;
                 };
       } & Omit<InputBaseFieldConfig, "endAdornment">)
-    | { type: "block"; block: BlockInterface }
-    | SingleFileFormFieldConfig
-    | MultiFileFormFieldConfig
+    | ({
+          type: "asyncSelectFilter";
+          name: string;
+          queryField: string; //TODO improve typing, use something similar to UsableFields<T>;
+          rootQuery: string;
+          labelField?: string;
+      } & Omit<InputBaseFieldConfig, "endAdornment">)
+    | { type: "block"; name: keyof T; block: BlockInterface }
+    | ({ type: "fileUpload"; multiple?: false; name: keyof T; maxFiles?: 1; download?: boolean } & Pick<
+          Partial<FinalFormFileUploadProps<false>>,
+          "maxFileSize" | "readOnly" | "layout" | "accept"
+      >)
+    | ({ type: "fileUpload"; multiple: true; name: keyof T; maxFiles?: number; download?: boolean } & Pick<
+          Partial<FinalFormFileUploadProps<true>>,
+          "maxFileSize" | "readOnly" | "layout" | "accept"
+      >)
 ) & {
-    name: keyof T;
```

Comment:
Had to move this into the field types, as asycnSelectFilter can't use `keyof T` (because it is a virtual field)